kotlinx-coroutines-core/common/src/flow/operators/Lint.kt

```diff
@@ -79,4 +80,61 @@ public fun FlowCollector<*>.cancel(cause: CancellationException? = null): Unit =
     replaceWith = ReplaceWith("currentCoroutineContext()")
 )
 public val FlowCollector<*>.coroutineContext: CoroutineContext
-    get() = noImpl()
\ No newline at end of file
+    get() = noImpl()
+
+@Deprecated(
+    message = "SharedFlow never completes, so this operator has no effect.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("this")
+)
+@InlineOnly
+public inline fun <T> SharedFlow<T>.catch(noinline action: suspend FlowCollector<T>.(cause: Throwable) -> Unit): Flow<T> =
+    (this as Flow<T>).catch(action)
+
+@Deprecated(
+    message = "SharedFlow never completes, so this operator has no effect.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("this")
+)
+@InlineOnly
+public inline fun <T> SharedFlow<T>.retry(
+    retries: Long = Long.MAX_VALUE,
+    noinline predicate: suspend (cause: Throwable) -> Boolean = { true }
+): Flow<T> =
+    (this as Flow<T>).retry(retries, predicate)
+
+@Deprecated(
+    message = "SharedFlow never completes, so this operator has no effect.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("this")
+)
+@InlineOnly
+public inline fun <T> SharedFlow<T>.retryWhen(noinline predicate: suspend FlowCollector<T>.(cause: Throwable, attempt: Long) -> Boolean): Flow<T> =
+    (this as Flow<T>).retryWhen(predicate)
+
+@Deprecated(
+    message = "SharedFlow never completes, so this terminal operation never completes.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("awaitCancellation()")
```

Comment:
`awaitCancellation` seems to be unintended

Reply:
I really cannot figure out a better replacement. We can leave it without a replacement, though.